src/object.rs

```diff
@@ -50,6 +50,14 @@ pub trait INSObject : Any + Sized + Message {
             Id::from_retained_ptr(obj)
         }
     }
+    
+    fn alloc() -> Id<Self> {
+        let cls = Self::class();
+        unsafe {
+            let obj: *mut Self = msg_send![cls, alloc];
+            Id::from_retained_ptr(obj)
```

Comment:
Thanks for the feedback! Your comments have convinced me create an initializer that does the allocation and initialization at the same time, like Object::new() does. So I withdraw my pull request. 

I do have some questions about a couple of things just to further my understanding. You wrote 

> Since it returns an Id<Self>, it'll automatically be dealloc'd later even if you never initialized it, so maybe it'd be better to return *mut Self instead.

I'm not sure why whether or not the object has been initialized should change the return type. The pointer returned from [NSObject alloc] would still need to be managed as a smart pointer, I think. I think I would want the object to be dealloc'ed even if it was not initialized. Though I would not expected anyone to actually just allocate without immediately initializing.

On returning an un-initialized object, you wrote:

> returning an un-initialized object is something I've never done in Objective-C

I've been away from Objective-C for a while so maybe my understanding is out of date, but it was a common idiom to `alloc` and `init` object  on the same line as in `[[MyObject alloc] init]`. For example, see the [Objective-C Wikipedia page in the "Instantiation" section](https://en.wikipedia.org/wiki/Objective-C#Instantiation). The example there of using a custom initializer is what I was trying to support by making `alloc` a separate function. The inner `alloc` does return an un-initialized object, but it was idiomatic to immediately initialize the object returned from `alloc`.

In my particular case, I'm going to write a function `NSWindow::newWithContentRect:...` that will alloc and init the window in the same function. So I don't need this change after all. 

Rob